### PR TITLE
Add http to https redirect

### PR DESCRIPTION
**Pull request by goruha**
# What
* Added variable that leave port 80 proxy to backends when https enabled

## Why
* Sometimes it's nice to automatically issue a redirect from http to https.

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -22,6 +22,11 @@ variable "loadbalancer_certificate_arn" {
   default = ""
 }
 
+variable "http_listen_always" {
```
Comment (osterman) [inline variables.tf]: rename to `http_listener_enabled`

**Comment by osterman** [comment]
-[ ] rebase

File: variables.tf (hunk 1 above)
Comment (osterman) [reply variables.tf]: Actually, I am a bit confused. Based on the title of the PR, I assume this has to do with TLS redirects. If that's what this does, then perhaps rename to `https_redirect`

Comment (goruha) [reply variables.tf]: Nope. 
By previously we had
```
if empty(var.loadbalancer_certificate_arn) {
 listen https
} else {
  listen http
} 
```

`http_listen_always` makes listen `http` even if `loadbalancer_certificate_arn` specified.

```
if not_empty(var.loadbalancer_certificate_arn) {
 listen https
}

if var.http_listener_enabled || empty(var.loadbalancer_certificate_arn)  {
  listen http
} 
```

Then backend will make redirect.

File: main.tf [hunk 2]
```diff
@@ -496,7 +496,7 @@ resource "aws_elastic_beanstalk_environment" "default" {
   setting {
     namespace = "aws:elb:listener"
     name      = "ListenerEnabled"
-    value     = "${var.loadbalancer_certificate_arn == "" ? "true" : "false"}"
+    value     = "${var.http_listen_always || var.loadbalancer_certificate_arn == "" ? "true" : "false"}"
```
Comment (osterman) [inline main.tf]: I still don't like the name. Let's rename to `http_listener_enabled`

Comment (goruha) [reply main.tf]: Done

**Pull request merged**